microsoft/CBL-Mariner
# feat: Allow new builds of Mariner to generate grub.cfg using grub2-mkconfig using imageconfig EnableGrubMkconfig

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*

---

###### Summary 

What does the PR accomplish, why was it needed?
This change allows new builds of Mariner to generate grub.cfg using grub2-mkconfig using EnableGrubMkconfig
- Introduce new image config which enables mkconfig by default.

###### Change Log  



- Allow new builds of Mariner to generate grub.cfg using grub2-mkconfig using KernelCommandLine.EnableGrubMkconfig
- Introduce new image config which enables mkconfig by default.
- Add imagecfg validation testing to ensure that images that enable mkconfig-boot by default include the needed utilities from grub2.rpm
- set $prefix in ESP partition grub.cfg which is analogous to $bootprefix in legacy boot cases; ensures grubenv is properly sourced by mkconfig-based grub.cfgs

###### Does this affect the toolchain?  



NO

###### Test Methodology

- locally; booted newly built image, observe that it boots/works the same, that grub.cfg is generated by mkconfig. 
- built an image with EnableGrubMkconfig: false. Observe that it builds, boots, has the legacy grub.cfg
- attempted building an image with EnableGrubMkconfig: true, but no grub2 package installed. Build stops in the tools, since the configuration is invalid.
- build an image with EnableGrubMkconfig: true. Observe that it builds, boots, has the mkconfig-based grub.cfg
- check behavior of grub2-editenv, see that changes to /boot/efi/grubenv are used by the generated grub.cfg
- Local build, with other related mkconfig changes cherry-picked into cameronbaird/3.0-mkconfig-integration
